coffeecard/CoffeeCard.Library/Utils/ProductUserGroupUtils.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+
+public class ProductUserGroupUtils
+{
+    /// <summary>
+    /// Determines whether a product is a perk or not.
+    /// A product is a perk if it is not available for regular customers.
+    /// </summary>
+    /// <param name="context">The DbContext to use for querying the database.</param>
+    /// <param name="product">The product to check.</param>
+    /// <returns>True if the product is a perk, false otherwise.</returns>
+    public static bool isPerk(DbContext context, Product product)
```

Comment:
C# has the neat functionality of Extension Methods. I think it makes sense to change your function to become an extension method. See https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods